src/Microsoft.Extensions.Diagnostics.HealthChecks/HealthChecksBuilderAddCheckExtensions.cs

```diff
@@ -88,5 +88,32 @@ public static IHealthChecksBuilder AddCheck(this IHealthChecksBuilder builder, I
             builder.Services.AddSingleton<IHealthCheck>(check);
             return builder;
         }
+
+        /// <summary>
+        /// Adds a new health check as a transient dependency injected service with the provided type.
+        /// </summary>
+        /// <typeparam name="T">The health check implementation type.</typeparam>
+        /// <param name="builder">The <see cref="IHealthChecksBuilder"/>.</param>
+        /// <returns>The <see cref="IHealthChecksBuilder"/>.</returns>
+        /// <remarks>
+        /// This method will register a transient service of type <see cref="IHealthCheck"/> with the 
+        /// provided implementation type <typeparamref name="T"/>. Using this method to register a health
+        /// check allows you to register a health check that depends on transient and scoped services.
+        /// </remarks>
+        public static IHealthChecksBuilder AddCheck<T>(this IHealthChecksBuilder builder) where T : class
```

Comment:
Couldn't you do `where T : class, IHealthCheck`? Better than doing a runtime check